creator-node/src/services/stateMachineManager/stateMonitoring/StateMonitoringQueue.js

```diff
@@ -81,150 +85,130 @@ class StateMonitoringQueue {
+   * @param {Function<successfulJob, jobResult>} params.monitorStateJobSuccessCallback the function to call when a monitorState job succeeds
+   * @param {Function<failedJob>} params.monitorStateJobFailureCallback the function to call when a monitorState job fails
+   * @param {Function<jobResult>} findPotentialSyncsJobSuccessCallback the function to call when a findPotentialSyncs job succeeds
+   * @param {Function<jobResult>} findReplicaSetUpdatesJobSuccessCallback the function to call when a findReplicaSetUpdates job succeeds
+   * @param {Function<job>} params.processMonitorStateJob the function to call when processing a job from the queue to monitor state
+   * @param {Function<job>} params.processFindPotentialSyncsJob the function to call when processing a job from the queue to find sync requests that potentially need to be issued
+   * @param {Function<job>} params.processFindReplicaSetUpdatesJob the function to call when processing a job from the queue to find users' replica sets that are unhealthy and need to be updated
    */
-  registerQueueEventHandlersAndJobProcessor({
+  registerQueueEventHandlersAndJobProcessors({
     queue,
-    jobSuccessCallback,
-    jobFailureCallback,
-    processJob
+    monitorStateJobSuccessCallback,
+    monitorStateJobFailureCallback,
+    findPotentialSyncsJobSuccessCallback,
+    findReplicaSetUpdatesJobSuccessCallback,
+    processMonitorStateJob,
+    processFindPotentialSyncsJob,
+    processFindReplicaSetUpdatesJob
   }) {
     // Add handlers for logging
     queue.on('global:waiting', (jobId) => {
-      this.log(`Queue Job Waiting - ID ${jobId}`)
+      logger.info(`Queue Job Waiting - ID ${jobId}`)
     })
     queue.on('global:active', (jobId, jobPromise) => {
-      this.log(`Queue Job Active - ID ${jobId}`)
+      logger.info(`Queue Job Active - ID ${jobId}`)
     })
     queue.on('global:lock-extension-failed', (jobId, err) => {
-      this.logError(
+      logger.error(
         `Queue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
       )
     })
     queue.on('global:stalled', (jobId) => {
-      this.logError(`stateMachineQueue Job Stalled - ID ${jobId}`)
+      logger.error(`Queue Job Stalled - ID ${jobId}`)
     })
     queue.on('global:error', (error) => {
-      this.logError(`Queue Job Error - ${error}. Queuing another job...`)
+      logger.error(`Queue Job Error - ${error}`)
     })
 
     // Add handlers for when a job fails to complete (or completes with an error) or successfully completes
     queue.on('completed', (job, result) => {
-      this.log(
-        `Queue Job Completed - ID ${job?.id} - Result ${JSON.stringify(
-          result
-        )}. Queuing another job...`
+      logger.info(
+        `Queue Job Completed - ID ${job?.id} - Result ${JSON.stringify(result)}`
       )
-      if (result?.jobFailed) {
-        jobFailureCallback(queue, job)
-      } else {
-        jobSuccessCallback(queue, job, result)
+      switch (job?.name) {
```

Comment:
we should def have a stateMachineManager readme that explains the diff queues and job types, and how they all connect to one another

potentially also a longer block comment at top of this file, or just a comment pointing to README

Reply:
definitely -- made a note to work on a readme later and link to it in index

Reply:
sick